numpy/core/_methods.py

```diff
@@ -189,10 +202,13 @@ def _var(a, axis=None, dtype=None, out=None, ddof=0, keepdims=False):
     # Compute the mean.
     # Note that if dtype is not of inexact type then arraymean will
     # not be either.
-    arrmean = umr_sum(arr, axis, dtype, keepdims=True)
+    arrmean = umr_sum(arr, axis, dtype, keepdims=True, where=where)
+    # The shape of rcount has to match arrmean to not change the shape of out
+    # in broadcasting. Otherwise, it cannot be stored back to arrmean.
+    div = (rcount if rcount.ndim == 0 else rcount.reshape(arrmean.shape))
```

Comment:
I think this would be clearer as a regular `if` statement, with a comment for each branch.

In particular, I think it's worth drawing attention to the fact that `rcount.ndim == 0` is the optimized case from `_count_reduce_items`

Reply:
Good suggestion, I will update it accordingly.